pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml

```diff
@@ -55,6 +55,9 @@ spec:
             - "--leader-election-renew-deadline={{ .LeaderElectionRenewDeadline }}"
             - "--leader-election-retry-period={{ .LeaderElectionRetryPeriod }}"
             - "--extra-create-metadata=true"
+          {{ if .VolumeGroupSnapshotSupported }}
```

Comment:
I think the following is possible, which may cause some confusion:

1. the external-snapshotter controller does not support the VolumeGroupSnapshots (old version)
2. a future version of the external-snapshotter sidecar is used that enables VolumeGroupSnapshots by default

This would cause a missing `--enable-volume-group-snapshots=false` option, and the external-snapshotter sidecar fails to start.

Mixing external-snapshotter controller (installed seperately) and sidecar version might be a corner case, but it looks easy to prevent this issue.

Reply:
@nixpanic am not able to get what exact problem can happen because of this check. i have added this check to ensure that we don't break anything and work as expected. if the CRDs are present we will set the `--enable-volume-group-snapshots=true` by default and users will still have the option to disable it if they want

Reply:
ha okay now i got it but i think this is a very very rare case and i dont expect that to happen if it happens the user need to downgrade the external snapshotter to the lower version. Again that is breaking change from the external snapshotter repo and documenting what the user should follow.

Again handling all the corner cases is very difficult to handle.